### ISSUE TITLE
Upon receiving a workspace/symbol request, the server panic if Capabilities.WorkspaceSymbolClientCapabilities.symbolKind isn't set by the client during initialize

### ISSUE THREAD

**Issue by shellwhale** [hashicorp/terraform-ls#1349]
### Language Server Version

v0.31.4

### Terraform Version

Terraform v1.5.4 on windows_amd64, Terraform v1.5.4 on linux_amd64

### Client Version

`terraform-ls/internal/langserver/handlers/workspace_symbol_test.go@TestLangServer_workspace_symbol_basic`

### Terraform Configuration

```terraform
# This the content in the workspace_symbol_test.go, but it's irrelevant to the bug.
provider "github" {}
```

### Steps to Reproduce

1. Open `terraform-ls/internal/langserver/handlers/workspace_symbol_test.go`
2. Go to the `TestLangServer_workspace_symbol_basic` function
3. Remove the whole "symbol" property (delete lines 44 included to 55 included)
4. Run the test `go test -timeout 30s -run ^TestLangServer_workspace_symbol_basic$ github.com/hashicorp/terraform-ls/internal/langserver/handlers`

It is worth mentioning that the error also occurs upon receiving a `textDocument/documentSymbol` request. There is also a test for it in `terraform-ls/internal/langserver/handlers/symbols_test.go`. The error is reproducible by removing lines 44 (included) to 58 (included).

### Expected Behavior

Expected some form of error handling to prevent the server crash inferred by the client and inform the user that some values should be set. Or have some default values that prevent the server crash. I'm not familiar enough with the Language Server Protocol Specification to know which one is the way to go.

### Actual Behavior

The server panics with the following message to the console :

`panic: runtime error: invalid memory address or nil pointer dereference`

### Gist

https://gist.github.com/shellwhale/a7d2f84e92838d4b1e4a01d921eebc99

### Help Wanted

- [ ] I'm interested in contributing a fix myself

### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by radeksimko** [comment]
Hi @shellwhale 
While panic isn't a desirable behaviour and is technically a violation of the LSP spec, I am curious if this in practice affects any users (i.e. is reproducible outside of a test)?

The [spec says](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#workspace_symbol) the following about the optional `valueSet`:

> 		 * If this property is not present the client only supports
>		 * the symbol kinds from `File` to `Array` as defined in
>		 * the initial version of the protocol.

We'll have to find what these are and maybe encode these kinds into a slice of defaults we can then pass in cases where `valueSet` and/or `symbolKind` aren't present.

I can look into a patch but I'd still like to establish the real-world impact.

**Comment by shellwhale** [comment]
> (i.e. is reproducible outside of a test)

Well, I was personally impacted! I was trying out to develop a client for the first time and came across the issue. I lost quite a bit of time wondering what could I be doing wrong.

I did not include my own client code in the issue for the sake of simplicity, I used the test so that the bug is easily reproducible.

I saw the symbolKind values are [here in the spec](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#symbolKind), so in our case a default slice like you said with values `1` and `18` would be sufficient I think.

I also just noticed that they are defined [here](https://github.com/hashicorp/terraform-ls/blob/e24f68ae2ebabc3fecf25aafdc39a5f4c89d49ef/internal/protocol/protocol.go#L6071C5-L6071C5) 

```go
var defaultSymbols protocol.SymbolKind = []int{protocol.File, protocol.Array}
```

I'd gladly make a PR for this but I'm not sure yet where we should put the default slice into.

**Comment by radeksimko** [comment]
Understood, to clarify - I wasn't implying this isn't worth fixing, I am just trying to assess the severity/impact of the bug so we can prioritise it alongside other work.

Also thank you for the report and for finding the bug.

> I'd gladly make a PR for this but I'm not sure yet where we should put the default slice into.

That would be most appreciated, thanks for the offer!

> ... is that what we are looking for?
> ```go
> var defaultSymbols protocol.SymbolKind = []protocol.SymbolKind{protocol.File, protocol.Array}
> ```

Yes, pretty much. The way I would interpret the comment in the spec

> symbol kinds from `File` to `Array` as defined in the initial version of the protocol

is all kinds between `File` and `Array`, i.e.

```go
var defaultSymbols = []protocol.SymbolKind{
	protocol.File,
	protocol.Module,
	protocol.Namespace,
	protocol.Package,
	protocol.Class,
	protocol.Method,
	protocol.Property,
	protocol.Field,
	protocol.Constructor,
	protocol.Enum,
	protocol.Interface,
	protocol.Function,
	protocol.Variable,
	protocol.Constant,
	protocol.String,
	protocol.Number,
	protocol.Boolean,
	protocol.Array,
}
```

**Comment by jpaju** [comment]
Hi! I'm encountering this issue consistently when using the [Helix editor](https://helix-editor.com/). This makes using the LSP quite hard because it will crash on **every** `workspace/symbol` and `textDocument/documentSymbol`, disabling all other LSP functionality.

Here are the requests from the client that crash the LSP:
```json
{
  "jsonrpc": "2.0",
  "method": "workspace/symbol",
  "params": { "query": "" },
  "id": 4
}
```

```json
{
  "jsonrpc": "2.0",
  "method": "textDocument/documentSymbol",
  "params": {
    "textDocument": {
      "uri": "file:<filepath>.tf"
    }
  },
  "id": 1
}
```

**Comment by dbanck** [comment]
We just released a new version of the language server [0.32.7](https://github.com/hashicorp/terraform-ls/releases/tag/v0.32.7). This fixes the reported bug.

Thank you for the report @shellwhale
